opentelemetry-api/src/logs/record.rs

```diff
@@ -30,6 +30,20 @@ pub struct LogRecord {
     pub attributes: Option<Vec<(Key, AnyValue)>>,
 }
 
+impl Default for LogRecord {
+    fn default() -> Self {
+        LogRecord {
+            timestamp: None,
+            observed_timestamp: SystemTime::now(),
```

Comment:
I wonder instead of using the `SystemTime` here I wonder if it makes sense to provide a generic method/struct to get current time here. And then we can allow users to provide a `dummy clock` and avoid the cost. It also makes testing easier

Reply:
Discussed offline:
1. Log-Appenders should not populate TimeStamp, instead they should only do ObservedTimeStamp (and they don't really have to do it, as they can rely on OTel API/SDK to populate it automatically). This is because, log-appenders may not be the only layers, and depending on the number/type of other layerss, it may be impossible to obtain the "true" timestamp. Unfortunately, neither of the log crates (log, tracing), providers the true timestamp. So the best option is to modify the PR (https://github.com/open-telemetry/opentelemetry-rust/pull/1175) to remove populating timestamp from the appenders.
2. with the above point, there is no longer the perf concern of reading ::now() twice.